src/interface.ts

```diff
@@ -17,10 +18,19 @@ export interface PaginationLocale {
   page_size?: string;
 }
 
+export interface PaginationPageSizeChanger
+  extends Pick<SelectProps, 'showSearch'> {
```

Comment:
It's not a good choice to export props from SelectProps since it's decided by `selectComponentClass`. Origin design is coupling. We should not make this more coupling.

We can make `showSizeChanger` to support renderProps:

```tsx
<Pagination
  showSizeChanger={({ value, onChange }) => (
    <Select value={value} onChange={onChange} showSearch />
  )}
/>
```

Reply:
@zombieJ I don't see full picture of solution, can you be more specific? 
So show `showSizeChanger` would be a bool or react element?
What is the default behavior for pagination size changer, with search or without?
What about all other props that related to size changer?

Reply:
OK. Like this:

```tsx
interface PaginationProps {
  // ...
  showSizeChanger?: boolean | ({ value, onChange }) => ReactNode;
}
```

When user want more flexible, use render props to create the selection element.